src/main/java/com/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,165 @@
+package com.wolox.training.controllers;
+
+import com.wolox.training.exceptions.BookNotFoundException;
+import com.wolox.training.exceptions.UserNotFoundException;
+import com.wolox.training.exceptions.UserUsernameRepeatedException;
+import com.wolox.training.models.Book;
+import com.wolox.training.models.User;
+import com.wolox.training.repositories.BookRepository;
+import com.wolox.training.repositories.UserRepository;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.util.Optional;
+
+@Controller
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    /**
+     * save the received user
+     * @param user user to be saved
+     * @exception UserUsernameRepeatedException when the username is already in the database.
+     * @return the user saved
+     */
+    @Operation(summary = "Create a user")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Successfully created", content = @Content),
```

Comment:
In successful responses, the content is usually brought automatically from the signature of the method in the controller.

If one wanted to define the content in a personalized way, it could be done as follows:

`@ApiResponse(responseCode = "401", description = AuthControllerConstants
                    .VALIDATE_TOKEN_RESPONSE_401,
                    content = @Content(schema = @Schema(implementation = ErrorResponse.class),
                            mediaType = MediaType.APPLICATION_JSON_VALUE))`